### PR TITLE
Deploy CDK Pipeline and Clean Up Repo

### PR DESCRIPTION
**Pull request by noah-paige**
- Bugfix

### Detail
- Allow permissions and specify externalID to allow `ddk deploy` of CDK Pipeline
- Fix clean up/deletion of cloned CDK Pipeline code base once deployed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/cdkproxy/cdk_cli_wrapper.py [hunk 1]
```diff
@@ -69,9 +69,8 @@ def clone_remote_stack(pipeline, pipeline_environment):
     print('..................................................')
     role_arn = f'arn:aws:iam::{pipeline_environment.AwsAccountId}:role/dataallPivotRole'
     sts = boto3.client('sts')
```
Comment (LEUNGUU) [inline backend/dataall/cdkproxy/cdk_cli_wrapper.py:71]: It seems we don't need this anymore.

File: backend/dataall/cdkproxy/stacks/cdk_pipeline.py [hunk 2]
```diff
@@ -225,11 +225,12 @@ def __init__(
         super().__init__(scope, f"dataall-{{environment_id.title()}}", **kwargs)
         DDKApplicationStack(self, "DataPipeline-{pipeline.label}-{pipeline.DataPipelineUri}", environment_id)
 
+id = f"dataall-cdkpipeline-{pipeline.DataPipelineUri}pip"
```
Comment (LEUNGUU) [inline backend/dataall/cdkproxy/stacks/cdk_pipeline.py:228]: What does pip mean in {pipeline.DataPipelineUri}pip?

File: backend/dataall/cdkproxy/cdk_cli_wrapper.py (hunk 1 above)
Comment (dlpzx) [reply backend/dataall/cdkproxy/cdk_cli_wrapper.py:71]: I agree

File: backend/dataall/cdkproxy/stacks/cdk_pipeline.py (hunk 2 above)
Comment (dlpzx) [reply backend/dataall/cdkproxy/stacks/cdk_pipeline.py:228]: For the case of CDK-pipelines pipeline type we deploy 2 stacks: one with the repository and one from running CDK deploy on the repo. Their target is the same, except for the pip suffix

**Review by dlpzx — APPROVED** [review]
i will approve and merge and remove the sts directly in the multi-env branch

**Pull request merged**